### DISCUSSION TITLE
make kotlinAllFieldsOptional not affect generated *input* types, only output ones

### DISCUSSION THREAD

**Discussion by bystam** [Netflix/dgs-framework#1224] [Ideas]
Hello,

I opened an [issue](https://github.com/Netflix/dgs-codegen/issues/448) and a [PR](https://github.com/Netflix/dgs-codegen/pull/449) on the matter that input types codegened in Kotlin uses the same nullability detection as output types, which means that the `kotlinAllFieldsOptional` property makes all input fields in the generated types nullable, ignoring the type information in the schema.

The `kotlinAllFieldsOptional` is useful for output types in order to decide whether return values should be inlined or resolved specifically using a custom `@DgsData` method. But this, in my mind, is not true for input types. There is no need for all fields to be nullable in input types. And currently using `kotlinAllFieldsOptional` means we lose that type information, and have to force unwrap a bunch everywhere in our code.

What do you think of changing this behaviour, as suggested in the PR/issue?

**Comment by srinivasankavitha** [comment]
I just reviewed and approved the PR. This suggested change in behavior makes sense to me. 🤞 it doesn't break anyone.
